scylla-cdc-lib/src/test/java/com/scylladb/cdc/lib/AlterUpdateUdtIT.java

```diff
@@ -0,0 +1,144 @@
+    assertEquals("cdc$batch_seq_no", change.getSchema().getColumnDefinition(2).getColumnName());
+    assertEquals("cdc$deleted_column3", change.getSchema().getColumnDefinition(3).getColumnName());
+    assertEquals("cdc$deleted_elements_column3", change.getSchema().getColumnDefinition(4).getColumnName());
+    assertEquals("cdc$end_of_batch", change.getSchema().getColumnDefinition(5).getColumnName());
+    assertEquals("cdc$operation", change.getSchema().getColumnDefinition(6).getColumnName());
+    assertEquals("cdc$ttl", change.getSchema().getColumnDefinition(7).getColumnName());
+    assertEquals("column1", change.getSchema().getColumnDefinition(8).getColumnName());
+    assertEquals("column2", change.getSchema().getColumnDefinition(9).getColumnName());
+    assertEquals("column3", change.getSchema().getColumnDefinition(10).getColumnName());
+    assertEquals(ChangeSchema.CqlType.UDT, change.getSchema().getColumnDefinition(10).getBaseTableDataType().getCqlType());
+    assertEquals("UDT(alterupdateudtit.simple_udt){a INT, b VARCHAR}", change.getSchema().getColumnDefinition(10).getBaseTableDataType().toString());
+    assertEquals(
+        "FROZEN<UDT(alterupdateudtit.simple_udt){a INT, b VARCHAR}>"
+        , change.getSchema().getColumnDefinition(10).getCdcLogDataType().toString()
+    );
+  }
+
+  @Override
+  public void verifyRawChangeAfterAlter(RawChange change) {
+
+    assertEquals("cdc$stream_id", change.getSchema().getColumnDefinition(0).getColumnName());
+    assertEquals("cdc$time", change.getSchema().getColumnDefinition(1).getColumnName());
+    assertEquals("cdc$batch_seq_no", change.getSchema().getColumnDefinition(2).getColumnName());
+    assertEquals("cdc$deleted_column3", change.getSchema().getColumnDefinition(3).getColumnName());
+    assertEquals("cdc$deleted_elements_column3", change.getSchema().getColumnDefinition(4).getColumnName());
+    assertEquals("cdc$end_of_batch", change.getSchema().getColumnDefinition(5).getColumnName());
+    assertEquals("cdc$operation", change.getSchema().getColumnDefinition(6).getColumnName());
+    assertEquals("cdc$ttl", change.getSchema().getColumnDefinition(7).getColumnName());
+    assertEquals("column1", change.getSchema().getColumnDefinition(8).getColumnName());
+    assertEquals("column2", change.getSchema().getColumnDefinition(9).getColumnName());
+    assertEquals("column3", change.getSchema().getColumnDefinition(10).getColumnName());
+    assertEquals(ChangeSchema.CqlType.UDT, change.getSchema().getColumnDefinition(10).getBaseTableDataType().getCqlType());
+    assertEquals("UDT(alterupdateudtit.simple_udt){a INT, b VARCHAR, c INT}", change.getSchema().getColumnDefinition(10).getBaseTableDataType().toString());
+    assertEquals(
+        "FROZEN<UDT(alterupdateudtit.simple_udt){a INT, b VARCHAR, c INT}>"
+        , change.getSchema().getColumnDefinition(10).getCdcLogDataType().toString()
+    );
+  }
+
+  @Override
+  public Runnable createDatagenTask() {
```

Comment:
`data generator/generation` is a bit weird, could please use something that fit better: data writer, table writer, record insert ...